src/sparseml/pytorch/datasets/classification/_ffcv/__ffcv.py

```diff
@@ -0,0 +1,234 @@
+    """
+
+    IMAGENET_MEAN = np.array([0.485, 0.456, 0.406]) * 255
+    IMAGENET_STD = np.array([0.229, 0.224, 0.225]) * 255
+    DEFAULT_CROP_RATIO = 224 / 256
+
+    def __init__(
+        self,
+        dataset,
+        write_path: str = "data/ffcv.beton",
+        transform: Optional[Callable] = None,
+        num_workers: int = 16,
+        image_size: int = 224,
+        device=default_device(),
+    ):
+        """
+        Initialize the dataset.
+        """
+        super().__init__(
+            dataset=dataset,
+            write_path=write_path,
+        )
+        self.transform = transform
+
+        with self.switched_off_transform():
+            self._write(num_workers=num_workers)
+
+        self._image_size = image_size
+        self._device = device
+
+    def _write(self, num_workers: int = 16):
+        """
+        Writes the dataset to the write_path.
+        """
+        assert ffcv, "FFCV is not installed"
+        writer = DatasetWriter(
+            self.write_path,
+            {
+                "image": RGBImageField(),
+                "label": IntField(),
+            },
+            num_workers=num_workers,
+        )
+
+        LOGGER.info(f"Writing dataset to {self.write_path}")
+        writer.from_indexed_dataset(self.dataset)
+
+    @contextmanager
+    def switched_off_transform(self):
+        """
+        Context manager to switch off the transforms in original dataset,
+        and then switch it back on after the context is exited.
+        """
+        _old_transform = None
+        _old_target_transform = None
+
+        try:
+            # Switch off the transforms
+            if hasattr(self.dataset, "transform"):
+                _old_transform = self.dataset.transform
+                self.dataset.transform = None
+                LOGGER.info("Transforms switched off for original dataset.")
+
+            if hasattr(self.dataset, "target_transform"):
+                _old_transform = self.dataset.target_transform
+                self.dataset.transform = None
+                LOGGER.info("Target Transforms switched off for original dataset.")
+
+            # Give back control
+            yield
+
+        finally:
+            # restore the transforms
+            if _old_transform is not None and hasattr(self.dataset, "transform"):
+                self.dataset.transform = _old_transform
+                LOGGER.info("Transforms switched on for original dataset.")
+
+            if _old_target_transform is not None and hasattr(
+                self.dataset, "target_transform"
+            ):
+                self.dataset.target_transform = _old_target_transform
+                LOGGER.info("Target Transforms switched on for original dataset.")
+
+    def get_loader(
+        self,
+        num_workers: int = 16,
+        batch_size: int = 32,
+        distributed: int = 0,
+        in_memory: int = 0,
+        validation: bool = False,
```

Comment:
let's not default at least num workers and batch size, use judgement on the rest

Reply:
These are overwritten when called from the train script, but would like to sync and figure out the right way forward